packages/flutter_test/lib/src/_goldens_io.dart

```diff
@@ -210,9 +210,11 @@ Future<ComparisonResult> compareLists(List<int>? test, List<int>? master) async
       error: 'Pixel test failed, image sizes do not match.\n'
         'Master Image: ${masterImage.width} X ${masterImage.height}\n'
         'Test Image: ${testImage.width} X ${testImage.height}',
+        diffs: <String, Image>{
+          'masterImage': masterImage,
+          'testImage': testImage,
+        },
     );
-    masterImage.dispose();
-    testImage.dispose();
```

Comment:
This should be restored, we'll still want to make sure these are disposed of properly.

Reply:
These are disposed over here: https://github.com/flutter/flutter/blob/master/packages/flutter_test/lib/src/goldeComparisonResultns.dart#L397

A similar pattern is in https://github.com/flutter/flutter/blob/master/packages/flutter_test/lib/src/_goldens_io.dart#L265-L269 where the ComparisonResult is returned (with the included images) and not disposed there.

Reply:
> A similar pattern is in https://github.com/flutter/flutter/blob/master/packages/flutter_test/lib/src/_goldens_io.dart#L265-L269 where the ComparisonResult is returned (with the included images) and not disposed there.

They are disposed there. :)

https://github.com/flutter/flutter/blob/9c1665355d9b0deeec1be60b9e9137037defa877/packages/flutter_test/lib/src/_goldens_io.dart#L265-L274

> These are disposed over here: https://github.com/flutter/flutter/blob/master/packages/flutter_test/lib/src/goldeComparisonResultns.dart#L397

There appears to be a typo in this link, but I think you mean here:

https://github.com/flutter/flutter/blob/9c1665355d9b0deeec1be60b9e9137037defa877/packages/flutter_test/lib/src/goldens.dart#L390-L398

Which is correct! So in that case can you remove the other calls to dispose in `_goldens_io.dart`?

Reply:
Sorry, not sure why that link 404s, but yes, you found the right place.

I'd be happy to make other changes, but I'm not sure about removing the other disposes. The images are only disposed if they aren't returned in the `diffs`. The only other `dispose` would be these https://github.com/flutter/flutter/blob/9c1665355d9b0deeec1be60b9e9137037defa877/packages/flutter_test/lib/src/_goldens_io.dart#L273-L274 and in that case, if those lines are reached the `ComparisonResult` will not contain the `masterImage`/`testImage` in their diff.

I think everything is disposed exactly once.

Reply:
Ah I see, thanks for digging into the code with me on this. :)